### PR TITLE
Refactor/profile stats

### PR DESCRIPTION
**Pull request by binarious**
Removed complex profile calculation to a separate class.

### CODE REVIEW

File: src/app/helper/profile.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    delete meditations.timespan;
+    return meditations;
+  }
+
+  initializeTimespans(user) {
+    let today = timezone(user, moment());
+    let todayWithoutTime = timezone(user, moment()).startOf('day');
+    let tenDaysAgo = moment(todayWithoutTime).subtract(9, 'days');
+    let tenWeeksAgo = moment(todayWithoutTime).subtract(9, 'weeks');
```
Comment (olastor) [inline src/app/helper/profile.js:45]: do you mean `.substract(10, 'weeks').add(1, 'day')`? Same thing for the next line

File: src/app/helper/profile.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      const duration = this.getStartOfDuration(date, meditations.lastDay);
+
+      // only one day ago = consecutive day
+      if (duration.asDays() ===   1) {
+        meditations.currentConsecutiveDays++;
+
+        meditations = this.tenDaysBadge(meditations);
+      } else if (duration.asDays() > 1) {
+        // more than one day ago = reset consecutive days
+        meditations.currentConsecutiveDays = 1;
```
Comment (olastor) [inline src/app/helper/profile.js]: When a user has no meditation entry today or the day before, `meditations.currentConsecuteDays` will still be 1. I think consecutive days should mean if you have no recent meditation log, then this means 0 consecutive days.

**Review by olastor — CHANGES_REQUESTED** [review]
One thing which is maybe worth of opening a new issue: Sessions which 'overflow' a day (i.e. 60min from 23:30 to 00:30 the following day) will be listed only on the day they were started. Maybe this could cause problems for users from different timezones.

**Comment by binarious** [comment]
@olastor please re-check. Does it work for you?

File: src/app/helper/profile.js (hunk 1 above)
Comment (binarious) [reply src/app/helper/profile.js:45]: No, because the current week and month is included..

**Comment by binarious** [comment]
@olastor It works for me. Can you check, too, please?

**Review by olastor — APPROVED** [review]

**Pull request merged**